projects/swimlane/ngx-charts/src/lib/box-chart/box.component.ts

```diff
@@ -0,0 +1,381 @@
+        [orientation]="orientation"
+        [name]="gradientId"
+        [stops]="gradientStops"
+      />
+      <svg:mask [attr.id]="maskLineId">
+        <svg:g>
+          <rect height="100%" width="100%" fill="white" fill-opacity="1" />
+          <path class="bar" [attr.d]="boxPath" fill="black" fill-opacity="1" />
+        </svg:g>
+      </svg:mask>
+    </svg:defs>
+    <svg:g>
+      <svg:path
+        class="bar"
+        role="img"
+        tabIndex="-1"
+        [class.active]="isActive"
+        [class.hidden]="hideBar"
+        [attr.d]="boxPath"
+        [attr.stroke]="strokeColor"
+        [attr.stroke-width]="strokeWidth"
+        [attr.aria-label]="ariaLabel"
+        [attr.fill]="hasGradient ? gradientFill : fill"
+        (click)="select.emit(data)"
+      />
+      <svg:line
+        class="bar-line"
+        [class.hidden]="hideBar"
+        [attr.x1]="lineCoordinates[0].v1.x"
+        [attr.y1]="lineCoordinates[0].v1.y"
+        [attr.x2]="lineCoordinates[0].v2.x"
+        [attr.y2]="lineCoordinates[0].v2.y"
+        [attr.stroke]="strokeColor"
+        [attr.stroke-width]="strokeWidth"
+        [attr.mask]="maskLine"
+        fill="none"
+      />
+      <svg:line
+        class="bar-line"
+        [class.hidden]="hideBar"
+        [attr.x1]="lineCoordinates[1].v1.x"
+        [attr.y1]="lineCoordinates[1].v1.y"
+        [attr.x2]="lineCoordinates[1].v2.x"
+        [attr.y2]="lineCoordinates[1].v2.y"
+        [attr.stroke]="strokeColor"
+        [attr.stroke-width]="strokeWidth"
+        fill="none"
+      />
+      <svg:line
+        class="bar-line"
+        [class.hidden]="hideBar"
+        [attr.x1]="lineCoordinates[2].v1.x"
+        [attr.y1]="lineCoordinates[2].v1.y"
+        [attr.x2]="lineCoordinates[2].v2.x"
+        [attr.y2]="lineCoordinates[2].v2.y"
+        [attr.stroke]="strokeColor"
+        [attr.stroke-width]="strokeWidth"
+        fill="none"
+      />
+      <svg:line
+        class="bar-line"
+        [class.hidden]="hideBar"
+        [attr.x1]="lineCoordinates[3].v1.x"
+        [attr.y1]="lineCoordinates[3].v1.y"
+        [attr.x2]="lineCoordinates[3].v2.x"
+        [attr.y2]="lineCoordinates[3].v2.y"
+        [attr.stroke]="strokeColor"
+        [attr.stroke-width]="strokeWidth"
+        fill="none"
+      />
+    </svg:g>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class BoxComponent implements OnChanges {
+  @Input() strokeColor: string;
+  @Input() strokeWidth: number;
+  @Input() fill: string;
+  @Input() data: IBoxModel;
+  @Input() width: number;
+  @Input() height: number;
+  @Input() x: number;
+  @Input() y: number;
+  @Input() lineCoordinates: [IVector2D, IVector2D, IVector2D, IVector2D];
+  @Input() orientation: string;
+  @Input() roundEdges: boolean = true;
+  @Input() gradient: boolean = false;
+  // TODO: Replace by IColorGradient Interface.
+  @Input() gradientStops: Array<{ offset: number; color: string; opacity: number }>;
```

Comment:
Anywhere you're referencing `IColorGradient`, please use https://github.com/swimlane/ngx-charts/blob/6d591944a7eaa2942b4b4c39c9d3526d28ceb1b9/projects/swimlane/ngx-charts/src/lib/common/types/gradient.interface.ts#L1